src/api/app.py

```diff
@@ -603,6 +603,24 @@ async def check_quota(
     return {}
 
 
+async def generate_thread_name(query: str) -> str:
+    """
+    Generate a descriptive name for a chat thread based on the user's query.
+
+    Args:
+        query: The user's initial query in the thread
+
+    Returns:
+        A concise, descriptive name for the thread
+    """
+    try:
+        prompt = f"Generate a concise, descriptive title (max 50 chars) for a chat conversation that starts with this query:\n{query}\nReturn strictly the title only, no quotes or explanation."
```

Comment:
This looks good! for better reliability we can use langchain structured outputs, here how we could implement this:
```python
from pydantic import BaseModel, Field

class ChatTitle(BaseModel):
    title: str = Field(max_length=50, description="Concise chat title")

prompt = f"Generate a concise, descriptive title for a chat conversation that starts with this query:\n{query}"

response = await HAIKU.with_structured_output(ChatTitle).ainvoke(prompt)
return response.title
```